### PR TITLE
[WTF-1986] Generate types for metadata attribute properties

### PR DESCRIPTION
**Pull request by weirdwater**
## Checklist

-   Contains unit tests ✅
-   Contains breaking changes ❌
-   Compatible with: MX 🔟 (After next release of typings package)
-   Did you update version and changelog?  ✅
-   PR title properly formatted (`[XX-000]: description`)? ✅

## This PR contains

-   [ ] Bug fix
-   [x] Feature
-   [ ] Refactor
-   [ ] Documentation
-   [ ] Other (describe)

## What is the purpose of this PR?

To lay the groundwork for offering cross-widget filtering and sorting of data sources. Enabling widget developers to build reusable and flexible filter controls.

## Relevant changes

This PR introduces the `AttributeMetaDataProperty` type. 

## What should be covered while testing?

That an attribute with the `isMetaData=true` flag results in a `AttributeMetaData` type.

### CODE REVIEW

File: packages/pluggable-widgets-tools/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## [Unreleased]
 
+### Added
+
+- We added the property type AttributeMetaData which exposes useful metadata for filtering and sorting datasources.
```
Comment (mlaponder) [inline packages/pluggable-widgets-tools/CHANGELOG.md:11]: I am not sure we should expose it in the changelog as we are not going public with this feature just yet

File: packages/pluggable-widgets-tools/src/typings-generator/__tests__/inputs/metadata-attribute.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+export const attributeMetaDataWebInput = `<?xml version="1.0" encoding="utf-8"?>
+<widget id="mendix.mywidget.MyWidget" needsEntityContext="true" offlineCapable="true"
+    pluginWidget="true" supportedPlatform="Web"
+    xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../xsd/widget.xsd">
+    <properties>
+        <propertyGroup caption="General">
+            <property key="data" type="dataSource">
```
Comment (mlaponder) [inline packages/pluggable-widgets-tools/src/typings-generator/__tests__/inputs/metadata-attribute.ts:8]: Two remarks here:
- This is actually an invalid configuration, as data source should be lower cased and the `isList=true` is required
- We should also test the `isLinked` datasource, we are not exporting that to the client so there shouldn't be a property typing generated for a linked data source

Comment (weirdwater) [reply packages/pluggable-widgets-tools/src/typings-generator/__tests__/inputs/metadata-attribute.ts]: Thanks, I should have picked up on the data source getting assigned an `any` type. I also updated the generator to skip linked data sources and added a test case to the datasource test.

File: packages/pluggable-widgets-tools/CHANGELOG.md (hunk 1 above)
Comment (weirdwater) [reply packages/pluggable-widgets-tools/CHANGELOG.md:11]: I can go either way on this. On the one hand it is nice to keep this feature close to the chest for now. On the other hand I enjoy the transparency of what changed in this particular version of the package. The generated experimental types are annotated with `@beta` so any user of it is warned...

As it is a single commit it is easily dropped.

File: packages/pluggable-widgets-tools/src/typings-generator/__tests__/inputs/metadata-attribute.ts (hunk 2 above)
Comment (mlaponder) [reply packages/pluggable-widgets-tools/src/typings-generator/__tests__/inputs/metadata-attribute.ts]: ✨

**Review by mlaponder — APPROVED** [review]

**Review by ancajuravle — APPROVED** [review]

**Pull request merged**